frontend/src/components/ColorPickerToolbar.jsx

```diff
@@ -35,7 +43,7 @@ function ColorPickerToolbar({primaryColor, setPrimaryColor, secondaryColor, setS
       <div className="color-display-box">
         <div
           className="color-display-first"
-          style={{ backgroundColor: primaryColor }}
+          style={{ backgroundColor: primaryColor || eyeDropperColor }}
```

Comment:
this isn't how we want to handle setting the color. We never want to be displaying a different value to the user than the one that will actully be used. Refactor out the eyeDropperColor state and just set the primaryColor directly when eyedropper is used.